[don't merge] ci: enable TEST_INTEGRATION_USE_SNAPSHOTTER

vvoland opened the pull request:
A draft PR to have the CI run integration tests with containerd integration enabled.

vvoland committed:
ci: enable TEST_INTEGRATION_USE_SNAPSHOTTER

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah commented:
@vvoland I did a quick rebase of this branch to get a fresh run of CI

thaJeztah committed:
c8d: use overlayfs snapshotter, and enable c8d-integration in Jenkins

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah commented:
- I cherry-picked https://github.com/moby/moby/pull/45322 🍿 

/cc @laurazard @rumpl

thaJeztah commented:
Seeing a lot of these failures; looks like something with restoring the frozen images? Haven't checked

```
TestDockerRegistryAuthHtpasswdSuite/TestLogoutWithExternalAuth
    docker_cli_logout_test.go:16: assertion failed: error is not nil: Error response from daemon: No such image: busybox:latest: [d5193e4374e5c] failed to download busybox
    --- FAIL: TestDockerRegistryAuthHtpasswdSuite/TestLogoutWithExternalAuth (0.67s)
```

rumpl commented on hack/make/.integration-test-helpers:
```diff
@@ -96,7 +96,7 @@ run_test_integration_suites() {
 				--raw-command \
 				-- go tool test2json -p "${pkgname}" -t ./test.main ${pkgtestflags}
-		); then exit 1; fi
+		)
```
GHA is green now even if tests fail :/

vvoland committed:
distribution: Add Tags

Add a function to return tags for the given repository reference. This
is needed to implement the `pull -a` (pull all tags) for containerd
which doesn't directly use distribution, but we need to somehow make an
API call to the registry to obtain the available tags.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
c8d/pull: Handle pull all tags

Use the distribution code to query the remote repository for tags and
pull them sequentially just like the non-c8d pull.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

rumpl commented:
Woop woop, we can close this now!!!!
